src/AntiDupl.NET.WPF/View/ProgressDialog.xaml.cs

```diff
@@ -1,27 +1,31 @@
-﻿using System;
-using System.Collections.Generic;
+﻿using System.ComponentModel;
+using System.Diagnostics;
 using System.Linq;
-using System.Text;
-using System.Threading.Tasks;
 using System.Windows;
-using System.Windows.Controls;
-using System.Windows.Data;
-using System.Windows.Documents;
-using System.Windows.Input;
-using System.Windows.Media;
-using System.Windows.Media.Imaging;
-using System.Windows.Shapes;
+using AntiDupl.NET.WPF.ViewModel;
 
-namespace AntiDupl.NET.WPF.View
+namespace AntiDupl.NET.WPF.View;
+
+/// <summary>
+///     Логика взаимодействия для ProgressDialog.xaml
+/// </summary>
+public partial class ProgressDialog : Window
 {
-    /// <summary>
-    /// Логика взаимодействия для ProgressDialog.xaml
-    /// </summary>
-    public partial class ProgressDialog : Window
+    public ProgressDialog()
+    {
+        InitializeComponent();
+    }
+
+    protected override void OnClosing(CancelEventArgs e)
     {
-        public ProgressDialog()
-        {
-            InitializeComponent();
-        }
+        base.OnClosing(e);
+
+        //find out, if the close button was pressed or if the window was closed by win.close()
```

Comment:
Is bad practise write anything in window file.

Reply:
It is not! It depends on what has to be done!